File: java/268A.java
// 268 A. Games

import java.util.*;

class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[][] teams = new int[n][2];

        for (int i = 0; i < n; i++) {
            teams[i][0] = sc.nextInt(); // home
            teams[i][1] = sc.nextInt(); // away
        }

        int count = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i != j && teams[i][0] == teams[j][1]) {
                    count++;
                }
            }
        }

        System.out.println(count);
        sc.close();
    }
}

File: cpp/268A.cpp
// 268 A. Games

#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<pair<int,int>> teams(n);

    for (int i = 0; i < n; i++) {
        cin >> teams[i].first >> teams[i].second; // home, away
    }

    int count = 0;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (i != j && teams[i].first == teams[j].second) {
                count++;
            }
        }
    }

    cout << count << endl;
    return 0;
}
